packages/server/src/lib/history.ts

```diff
@@ -1,48 +1,85 @@
 import {
     ArticleCounts,
     ArticlesViewedSettings,
+    ArticlesViewedByTagSettings,
     WeeklyArticleHistory,
     WeeklyArticleLog,
 } from '@sdc/shared/types';
 
+// From https://github.com/guardian/automat-client-v2/blob/master/src/contributions/lib/dates.ts#L4
 export const getMondayFromDate = (date: Date): number => {
-    const day = date.getDay() || 7;
-    if (day !== 1) {
-        date.setHours(-24 * (day - 1));
-    }
-    return Math.floor(date.getTime() / 86400000);
```

Comment:
this code had a bug which should have been fixed here as well, see https://github.com/guardian/automat-client-v2/pull/7